### PR TITLE
Fix obsolete XML comments

### PR DESCRIPTION
**Pull request by jmprieur**
Fixes XML comments which were still referencing AcquireTokenXXAsync methods.

### CODE REVIEW

**Comment by jmprieur** [comment]
@jennyf19 thanks for approving
I've added more comments in MsalError.cs as well to make the errors more actionable. In case you want to review :)

File: src/Microsoft.Identity.Client/MsalError.cs [hunk 1]
```diff
@@ -133,6 +147,10 @@ public static class MsalError
 
         /// <summary>
         /// Invalid authority type.
+        /// MSAL.NET does not know how to interact with the authority specified when the application was built (for instance MSAL.NET
+        /// does not support ADFS 2016 and earlier).
+        /// <para>Mitigation</para>
```
Comment (jennyf19) [inline src/Microsoft.Identity.Client/MsalError.cs]: Nit: I wouldn't mention this part on ADFS2016

File: src/Microsoft.Identity.Client/MsalError.cs [hunk 2]
```diff
@@ -144,29 +162,54 @@ public static class MsalError
...
 
         /// <summary>
         /// Invalid owner window type.
+        /// <para>What happens?</para>
+        /// You used <see cref="AcquireTokenInteractiveParameterBuilder.WithParentActivityOrWindow(object)"/>
+        /// but the parameter you passed is invalid.
+        /// <para>Remediation</para>
+        /// On .NET Standard, the expected object is an <c>Activity</c> on Android, a <c>UIViewController</c> on iOS,
+        /// a <c>NSWindow</c> on MAC, and a <c>IWin32Window</c> or <c>IntPr</c> on Windows.
+        /// If you are in a WPF application, you can use <c>ew WindowInteropHelper(wpfControl).Handle</c>
```
Comment (jennyf19) [inline src/Microsoft.Identity.Client/MsalError.cs]: ''ew'?

File: src/Microsoft.Identity.Client/MsalError.cs [hunk 3]
```diff
@@ -233,17 +305,29 @@ public static class MsalError
...
         /// </summary>
         public const string UnknownUserType = "unknown_user_type";
 
         /// <summary>
-        /// Unknown User.
+        /// <para>What happens</para>
+        /// You can get this error when using <see cref="IPublicClientApplication.AcquireTokenByUsernamePassword(System.Collections.Generic.IEnumerable{string}, string, System.Security.SecureString)"/>
+        /// The user is not known by the IdP
+        /// <para>Mitigation</para>
+        /// Inform the user. the login that the user provided might be incorrect (for instance empty)
```
Comment (jennyf19) [inline src/Microsoft.Identity.Client/MsalError.cs]: Nit: Capitol t on the

File: src/Microsoft.Identity.Client/MsalError.cs [hunk 4]
```diff
@@ -254,21 +338,43 @@ public static class MsalError
...
         public const string PasswordRequiredForManagedUserError = "password_required_for_managed_user";
 
         /// <summary>
         /// Request is invalid.
+        /// <para>What happens?</para>
+        /// This can happen because you are using a token acquisition method which is not compatible with the authority. For instance:
+        /// you called <see cref="IPublicClientApplication.AcquireTokenByUsernamePassword(System.Collections.Generic.IEnumerable{string}, string, System.Security.SecureString)"/>
+        /// but you used an authority ending with '/common' or '/consumers' as this requires a tenanted authority or '/organizations'.
+        /// <para>Mitigation</para>
+        /// Adjut the authority to the AcquireTokenXX method you use (don't use 'common' or 'consumers' with <see cref="IPublicClientApplication.AcquireTokenByUsernamePassword(System.Collections.Generic.IEnumerable{string}, string, System.Security.SecureString)"/>
```
Comment (jennyf19) [inline src/Microsoft.Identity.Client/MsalError.cs]: Adjust?

**Comment by jmprieur** [comment]
Thanks @jennyf19. I'll fix your suggestions.
For the todo items, do you mean the one I raised? if yes, we need to decide if the errors can emitted?if not we shoud remove them?

**Comment by jennyf19** [comment]
@jmprieur I think we should remove them if they are not used anywhere

**Comment by jmprieur** [comment]
@jennyf19 I'll raise an issue so to remove the couple of errors that don't seem to be emitted (we need to check that this is indeed the case)

**Comment by bgavrilMS** [comment]
Build passed

**Comment by jmprieur** [comment]
Thanks @bgavrilMS

**Pull request merged**